lib/rule_engine/ast.py

```diff
@@ -55,6 +55,10 @@ def _assert_is_numeric(*values):
 	if not all(map(is_numeric, values)):
 		raise errors.EvaluationError('data type mismatch (not a numeric value)')
 
+def _assert_is_string(*values):
+	if not all(map(isinstance, values, [str])):
```

Comment:
i didn't follow your of defining a helper function in types here, it wasn't going to be used elsewhere and this was succint